### PR TITLE
Scan periscope gradient data for large changes and send email

### PR DESCRIPTION
**Pull request by jeanconn**
## Description

Scan periscope gradient data for large changes and send email

### Unit tests

- [x] No unit tests

### Functional tests

I ran this with no data file, and confirmed it sends an email with the list of instances.  I then removed the last instance that started with a dwell with a science observation and it sends an email that includes obsid 27506.
```
Discontinuities found in periscope gradient data.
Check V&V for these science observations.
date | msid | AOPCADMD | AOACASEQ | COBSRQID | manvr_obsid
2022:293:16:26:47.422 | OOBAGRD6 | NPNT | KALM | 27506.0 | 27506
2022:293:16:27:20.222 | OOBAGRD6 | NPNT | KALM | 27506.0 | 27506
2022:295:16:47:24.231 | OOBAGRD6 | STBY | AQXN | 0.0 | 27506
2022:295:16:47:57.031 | OOBAGRD3 | STBY | AQXN | 0.0 | 27506
2022:295:17:21:50.632 | OOBAGRD6 | STBY | BRIT | 0.0 | 27506
2022:295:17:21:50.632 | OOBAGRD3 | STBY | BRIT | 0.0 | 27506
2022:295:17:22:23.432 | OOBAGRD3 | STBY | BRIT | 0.0 | 27506
2022:295:17:22:23.432 | OOBAGRD6 | STBY | BRIT | 0.0 | 27506
2022:295:17:43:42.632 | OOBAGRD3 | NSUN | AQXN | 0.0 | 27506
```

### CODE REVIEW

**Review by taldcroft — APPROVED** [review]

File: ska_trend/bad_periscope_gradient/periscope_update.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    for msid in msids:
+        dat = fetch.Msid(msid, start, filter_bad=True)
+        dat.interpolate(dt=sampling)
+        diff = np.diff(dat.vals)
+        nok = np.abs(diff) > quantization * quant_mult
+
+        for hit in np.where(nok)[0]:
+            event = {
+                "time": dat.times[hit],
+                "date": CxoTime(dat.times[hit]).date,
```
Comment (jeanconn) [inline ska_trend/bad_periscope_gradient/periscope_update.py:75]: Fix DRY for dat.times[hit]

File: ska_trend/bad_periscope_gradient/periscope_update.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                "msid": msid,
+            }
+            local_manvrs = events.manvrs.filter(
+                start=dat.times[hit] - (7 * 86400), stop=dat.times[hit] + (1 * 86400)
+            )
+            extra_msids = ["COBSRQID", "AOPCADMD", "AOACASEQ"]
+            extra_defaults = {"COBSRQID": 0, "AOPCADMD": "N/A", "AOACASEQ": "N/A"}
```
Comment (jeanconn) [inline ska_trend/bad_periscope_gradient/periscope_update.py:82]: Add a comment about why these have defaults.

File: ska_trend/bad_periscope_gradient/periscope_update.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            for extra_msid in extra_msids:
+                mdat = fetch.Msid(extra_msid, dat.times[hit] - 200, dat.times[hit] + 50)
+                # get last sample before hit
+                hit_idx = np.searchsorted(mdat.times, dat.times[hit])
+                if hit_idx == 0:
```
Comment (jeanconn) [inline ska_trend/bad_periscope_gradient/periscope_update.py:87]: Data is missing around really interesting things like bad gradient data, add a comment that explains this.

File: ska_trend/bad_periscope_gradient/periscope_update.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                    event.update({extra_msid: extra_defaults[extra_msid]})
+                else:
+                    event.update({extra_msid: mdat.vals[hit_idx - 1]})
+            last_manvr = None
```
Comment (jeanconn) [inline ska_trend/bad_periscope_gradient/periscope_update.py:91]: Not sure about the cleanest method to just find the last expected obsid for the commanded maneuver/dwell.  This is a little convoluted but seems to work robustly when there is bad or missing data (including kadi events based on telemetry).

**Pull request merged**